plugins/DigitalLibrary/resources/qml/SelectProjectPage.qml

```diff
@@ -61,13 +61,28 @@ Item
             id: createNewProjectButton
 
             text: "New Library project"
+            visible: createNewProjectButtonVisible && manager.userAccountCanCreateNewLibraryProject && (manager.retrievingProjectsStatus == DF.RetrievalStatus.Success || manager.retrievingProjectsStatus == DF.RetrievalStatus.Failed)
 
             onClicked:
             {
                 createNewProjectPopup.open()
             }
             busy: manager.creatingNewProjectStatus == DF.RetrievalStatus.InProgress
         }
+
+
+        Cura.SecondaryButton
+        {
+            id: upgradePlanButton
+
+            text: "Upgrade plan"
+            iconSource: UM.Theme.getIcon("LinkExternal")
+            visible: createNewProjectButtonVisible && !manager.userAccountCanCreateNewLibraryProject && (manager.retrievingProjectsStatus == DF.RetrievalStatus.Success || manager.retrievingProjectsStatus == DF.RetrievalStatus.Failed)
```

Comment:
Why does this need to depend on `DF.RetrievalStatus` and `manager.retrievingProjectsStatus`? Shouldn't it just depend on `manager.userAccountCanCreateNewLibraryProject`? If it fails to retrieve the list of projects, the last known information should count, right?

Reply:
To check for private projects we need to query the `/projects` API endpoint. So the button should be shown after the GET request for retrieving the private projects has finished.